tabu/src/tabu_search.cc

```diff
@@ -90,7 +92,8 @@ void TabuSearch::multiStartTabuSearch(long long timeLimitInMilliSecs,
     vector<vector<double>> C(bqp.nVars, vector<double>(bqp.nVars));
 
     for (long iter = 0; iter < numRestarts; iter++) {
-        if (useTimeLimit && (realtime_clock() - startTime) > timeLimitInMilliSecs) {
+        if ((bestSolutionQuality <= energyThreshold) ||
```

Comment:
Would it make sense to add this condition to the inner loop, in `simpleTabuSearch`? It probably would help break out earlier on huge problems? Not sure it's worth the mess. I'd at least document where the test for energy is done.